src/execute_linter.cc

```diff
@@ -13,55 +13,157 @@
 #include <streambuf>
 #include <string>
 #include <vector>
 
+#include "absl/strings/str_split.h"
 #include "absl/strings/string_view.h"
+#include "absl/strings/strip.h"
+#include "re2/re2.h"
 #include "src/lint_errors.h"
 #include "src/linter.h"
+#include "src/linter_options.h"
 #include "zetasql/base/status.h"
 #include "zetasql/base/status_macros.h"
 
 namespace zetasql::linter {
 
-namespace {
-// It runs all specified checks
-// "LinterOptions" parameter will be added in future.
-LinterResult RunChecks(absl::string_view sql) {
+// This function will parse "NOLINT"(<CheckName>) syntax from
+// a single line of comment.
+// If NOLINT usage errors occur, it counts as a lint
+// error and it will be returned in a result.
+LinterResult ParseNoLintSingleComment(absl::string_view line,
+                                      const absl::string_view& sql,
+                                      LinterOptions* options, int position) {
   LinterResult result;
-  result.Add(CheckLineLength(sql));
-  result.Add(CheckParserSucceeds(sql));
-  result.Add(CheckSemicolon(sql));
-  result.Add(CheckUppercaseKeywords(sql));
-  result.Add(CheckCommentType(sql));
-  result.Add(CheckAliasKeyword(sql));
-  result.Add(CheckTabCharactersUniform(sql));
-  result.Add(CheckNoTabsBesidesIndentations(sql));
+  std::map<std::string, ErrorCode> error_map = GetErrorMap();
+
+  const LazyRE2 kRegex = {"\\s*(NOLINT|LINT)\\s*\\(([a-z ,-]*)\\)\\s*(.*)\\s*"};
+  std::string type = "";
+  std::string check_names = "";
+  std::string lint_comment = "";
+  bool matched =
+      RE2::FullMatch(line, *kRegex, &type, &check_names, &lint_comment);
+
+  // Found enabling or disabling type of comment.
+  if (matched) {
+    check_names.erase(
+        std::remove_if(check_names.begin(), check_names.end(), ::isspace),
+        check_names.end());
+
+    std::vector<std::string> names = absl::StrSplit(check_names, ',');
+    for (std::string check_name : names) {
+      // The name inside of parantheses is stored in 'check_name'
+      // If it is not valid add error, otherwise enable/disable position
+      if (!error_map.count(check_name)) {
+        result.Add(ErrorCode::kNoLint, sql, position,
+                   absl::StrCat("Unkown NOLINT error category: ", check_name));
+      } else {
+        const ErrorCode& code = error_map[check_name];
+        if (type == "NOLINT")
+          options->Disable(code, position);
+        else
+          options->Enable(code, position);
+      }
+    }
+  }
   return result;
 }
 
-}  // namespace
-}  // namespace zetasql::linter
+// This function will parse "NOLINT"(<CheckName>) syntax from a sql file.
+// If NOLINT usage errors occur, it counts as a lint
+// error and it will be returned in a result.
+LinterResult ParseNoLintComments(absl::string_view sql,
+                                 LinterOptions* options) {
+  LinterResult result;
+  for (int i = 0; i < static_cast<int>(sql.size()); ++i) {
+    if (sql[i] == '\'' || sql[i] == '"') {
```

Comment:
Can we have regex for this loop as well?

Reply:
That would be super complicated regex. I think regex code would be much more hard to read. This part just ignores the part inside of string in few lines of code.